vendor/github.com/samuel/go-zookeeper/zk/dnshostprovider.go

```diff
@@ -0,0 +1,88 @@
+package zk
+
+import (
+	"fmt"
+	"net"
+	"sync"
+)
+
+// DNSHostProvider is the default HostProvider. It currently matches
+// the Java StaticHostProvider, resolving hosts from DNS once during
+// the call to Init.  It could be easily extended to re-query DNS
+// periodically or if there is trouble connecting.
+type DNSHostProvider struct {
```

Comment:
There's a reason why we made Mesos-DNS forcibly die in DC/OS if it can't connect to zk within a relatively short window. Killing / restarting the process always kills bad state. Also makes it so things like load balancers / spartan don't forward traffic to DNS nodes which are serving stale data which doesn't put up any other bad health signs. I'm a big fan of keeping the behavior simple / just kill Mesos-DNS and let it be restarted if things don't heal fairly quickly.